lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
Creating a pool when 300ish pools exist without this code takes 17 seconds, so adds up to +3 seconds per create with tons of pools.